DDBL Stock

In [ ]:
import pandas as pd

df = pd.read_csv("/content/subset.csv")
new_df = df.head(10)
print(new_df.tail(5))

         date symbol      open      high       low     close  volume
5  2023-04-10   DDBL  542.2906  547.2657  542.2906  546.4366  1740.0
6  2023-04-11   DDBL  546.4366  555.5577  545.6903  548.9241  2946.0
7  2023-04-12   DDBL  559.7036  579.3554  559.7036  579.3554  6973.0
8  2023-04-13   DDBL  590.8812  590.8812  565.5079  567.9955  6066.0
9  2023-04-16   DDBL  572.1415  572.1415  553.8993  567.9955  3507.0


1. Range

In [ ]:
volume = new_df['volume'].to_list() # dataframe lai list ma convert garim
max_vol = new_df['volume'].max() #max value
min_vol = new_df['volume'].min() #min value
range = max_vol-min_vol
print(f"Range: {range}") # range

Range: 5233.0


In [ ]:
# variance = new_df['volume'].var()
# print(variance)

2. Variance

In [ ]:
mean = new_df['volume'].sum()/len(volume)
print(f"Mean: {mean}")
sum =0
for x in volume:
  sum += (x-mean)**2
  # print(sum)
print(f"Sum: {sum}")
#variance = sigma(x-mean)^2/(N-1) for sample variance
variance = sum/(len(volume) - 1)
print(f"Sample Variance: {variance}")

Mean: 3745.3
Sum: 23644800.099999998
Sample Variance: 2627200.011111111


3. Standard Deviation

In [ ]:
sd = variance**0.5
print(f"Standard deviation: {sd}")

Standard deviation: 1620.863970575912


In [ ]:
z = (3507.0 - mean)/ sd
print(f"Z-score: {z}")

Z-score: -0.1470203572452347


Range = 5233, Mean =3745.3, Variance = 2627200.01, Standard deviation = 1620.8
Conclusion: The Z-score for the last day's volume, relative to the top 10 volumes, is -0.14, which suggests panic selling.